src/compas_viewer/components/slider.py

```diff
@@ -2,176 +2,112 @@
-        Additional keyword arguments for the action.
-
-    Attributes
-    ----------
-    action : callable
-        Action associated with the button click event.
-    slider : QtWidgets.QSlider
-        The actual slider widget.
-    value : float
-        The current value of the slider.
-    STYLE : str
-        Stylesheet for the visual appearance of the groove and handle of the slider.
-
-    See Also
-    --------
-    :class:`compas_viewer.layout.SidedockLayout`
-
-    References
-    ----------
-    :PySide6:`PySide6/QtWidgets/QSlider`
-
-    """
-
-    STYLE = """
-    QSlider::groove::horizontal {
-        border: 1px solid #cccccc;
-        background-color: #eeeeee;
-        height: 4px;
-    }
-    QSlider::handle:horizontal {
-        background-color: #ffffff;
-        border: 1px solid #cccccc;
-        border-radius: 6px;
-        height: 12px;
-        width: 12px;
-        margin: -6px 0;
-    }
-    """
 
+class Slider(QWidget):
     def __init__(
         self,
-        action: Callable,
-        value: int = 0,
-        min_value: int = 0,
-        max_value: int = 100,
-        step: int = 1,
-        title: Optional[str] = None,
-        annotation: Optional[str] = None,
-        interval: int = 1,
-        bgcolor: Optional[Color] = None,
-        stretch: int = 0,
-        **kwargs,
+        title: str = "Slider",
+        horizontal: Optional[bool] = True,
+        min_val: int = 0,
+        max_val: int = 100,
+        step: float = 0.5,
+        action: Callable = None,
+        tick_interval: Optional[float] = None,
+        starting_val: Optional[float] = None,
     ):
+        """
+        A customizable slider widget for Qt applications, supporting both horizontal and vertical orientations. This
+        widget displays the current, minimum, and maximum values and allows for dynamic user interaction.
+
+        Parameters
+        ----------
+        title : str, optional
+            Label displayed above the slider, defaults to "Slider".
+        horizontal : bool, optional
+            Orientation of the slider. True for horizontal, False for vertical. Defaults to True.
+        min_val : float, optional
+            Minimum value of the slider, defaults to 0.
+        max_val : float, optional
+            Maximum value of the slider, defaults to 100.
+        action : Callable, optional
+            Function to execute on value change. Should accept a single integer argument.
+        tick_interval : float, optional
+            Interval between tick marks. No ticks if None. Defaults to None.
+        starting_val : float, optional
+            Initial value of the slider, defaults to the minimum value.
+
+        Attributes
+        ----------
+        min_label : QLabel
+            Label displaying the minimum value.
+        max_label : QLabel
+            Label displaying the maximum value.
+        value_label : QLabel
+            Label displaying the current value prefixed by the title.
```

Comment:
These are not the attributes. The attributes are things that you assigned in `__init__` like `self.xxx = xxx`, in this case should be `title`, `action`, `horizon` ... `layout`, `h_layout`, `slider` ... if this list become too big, you should consider adding `_` in front of the ones which are purely for internal use, for example `_h_layout`. This means user don't have to know about this one. In `Attributes`  section you don't have to document the ones starts with `_`.